pyvista/core/datasetattributes.py

```diff
@@ -0,0 +1,317 @@
+"""Implements DataSetAttributes, which represents & manipulates datasets."""
+
+import numpy as np
+import pyvista.utilities.helpers as helpers
+from .pyvista_ndarray import pyvista_ndarray
+from pyvista.utilities.helpers import FieldAssociation
+import vtk
+from vtk.numpy_interface.dataset_adapter import VTKObjectWrapper, numpyTovtkDataArray
+
+
+class DataSetAttributes(VTKObjectWrapper):
+    """
+    Python friendly wrapper of vtk.DataSetAttributes.
+
+    Implement a dict like interface for interacting with vtkDataArrays.
+
+    Parameters
+        ----------
+        vtkobject : vtkFieldData
+            The vtk object to wrap as a DataSetAttribute,
+             usually an instance of vtkCellData, vtkPointData, vtkFieldData.
+
+        dataset : vtkDataSet
+            The vtkDataSet containing the vtkobject.
+
+        association : FieldAssociation
+            The array association type of the vtkobject.
+    """
+
+    def __init__(self, vtkobject, dataset, association):
+        """Initialize DataSetAttributes."""
+        super().__init__(vtkobject=vtkobject)
+        self.dataset = dataset
+        self.association = association
+
+    def __getitem__(self, key):
+        """
+        Implement get behaviour of [] operator.
```

Comment:
Let's be consistent with the docstring; first line shouldn't be empty.

Reply:
Fixed.